ibis/tests/expr/test_datatypes.py

```diff
@@ -545,3 +545,97 @@ def test_traversal_replace():
     result = dtype.replace(subs)
     expected = dt.Struct.from_tuples([('a', 'bool'), ('b', dt.Array('timestamp'))])
     assert result == expected
+
+
+def get_leaf_classes(op):
```

Comment:
Would it make sense to use this utility for actually defining the methods?

Reply:
Maybe, but let's see how it is to use the manually defined ones first.